packages/flutter_tools/lib/src/isolated/devfs_web.dart

```diff
@@ -1004,17 +1011,49 @@ Future<Directory> _loadDwdsDirectory(FileSystem fileSystem, Logger logger) async
 }
 
 String _stripBasePath(String path, String basePath) {
-  while (path.startsWith('/')) {
-    path = path.substring(1);
-  }
+  path = _stripLeadingSlashes(path);
   if (path.startsWith(basePath)) {
     path = path.substring(basePath.length);
   } else {
     // The given path isn't under base path, return null to indicate that.
     return null;
   }
+  return _stripLeadingSlashes(path);
+}
+
+String _stripLeadingSlashes(String path) {
   while (path.startsWith('/')) {
     path = path.substring(1);
   }
   return path;
 }
+
+String _stripTrailingSlashes(String path) {
+  while (path.endsWith('/')) {
+    path = path.substring(0, path.length - 1);
+  }
+  return path;
+}
+
+String _parseBasePathFromIndexHtml(File indexHtml) {
+  final String htmlContent = indexHtml.existsSync()
+      ? indexHtml.readAsStringSync()
+      : _kDefaultIndex;
+
+  final Document document = parse(htmlContent);
+  final Element baseElement = document.querySelector('base');
+  String baseHref = baseElement?.attributes == null ? null : baseElement.attributes['href'];
+
+  if (baseHref == null) {
+    baseHref = '';
+    print('WARNING: Please add a <base> element to your index.html file.');
```

Comment:
baseref is really an optional feature right? I would leave this out and we can update release notes instead.

As is this message does not give enough context for a user to know what/how to fix it

Reply:
Makes sense. Where do I update release notes?

Reply:
changelog on the wiki is a good place to start. We'll need to collect all of the index.html changes up offline and possibly ask users just to regenerate